chef/cookbooks/cpe_zoom/resources/cpe_zoom.rb

```diff
@@ -17,21 +15,110 @@
 
 default_action :run
 
-# Enforce Zoom Settings
-action :run do
-  zoom_prefs = node['cpe_zoom'].compact
-  unless zoom_prefs.empty?
-    zoom_prefs.each_key do |key|
-      next if zoom_prefs[key].nil?
+action_class do
+  WINDOWS_GENERAL_SETTINGS_MAP = {
+    "ZAutoSSOLogin" => "ForceLoginWithSSO",
+    "ZSSOHost" => "ForceSSOURL",
+    "ZAutoUpdate" => "EnableClientAutoUpdate",
+    "disableloginwithemail" => "DisableLoginWithEmail",
+    "nofacebook" => "DisableFacebookLogin",
+    "nogoogle" => "DisableGoogleLogin",
+  }.freeze
+
+  WINDOWS_CHAT_SETTINGS_MAP = {
+    "DisableLinkPreviewInChat" => "DisableLinkPreviewInChat",
+  }.freeze
+
+  WINDOWS_IGNORABLE_SETTINGS_MAP = [
+    "LastLoginType",
+    "forcessourl", # ZSSOHost is already mapped to ForceSSOURL
+  ].freeze
+
+  def zoom_prefs
+    node["cpe_zoom"].compact
+  end
+
+  def convert_to_registry_key(value)
+    if value == true
+      return { "data": 1, "type": :dword }
+    elsif value == false
+      return { "data": 0, "type": :dword }
+    else
+      return { "data": value.to_s, "type": :string }
+    end
+  end
+
+  def configure_windows
+    zoom_settings = {}
+    zoom_prefs.each do |key, value|
+      next if WINDOWS_IGNORABLE_SETTINGS_MAP.include?(key) # Skippable preferences on Windows
+
+      preference = WINDOWS_CHAT_SETTINGS_MAP[key]
+      if preference.nil? # If the preference isn't in WINDOWS_CHAT_SETTINGS_MAP, set General path.
+        zoom_settings[WINDOWS_GENERAL_SETTINGS_MAP.fetch(key, key)] = { "path": 'HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Zoom\Zoom Meetings\General', "name": WINDOWS_GENERAL_SETTINGS_MAP.fetch(key, key) }.merge(convert_to_registry_key(value))
```

Comment:
Have you run Rubocop against this source feature branch (using the `.ruboop.yml` from the root of our project repo)? 

It looks like this line is ~240chars (and our max `LineLength` for Rubocop is 120chars).

Reply:
done